src/components/login/__tests__/login.common.test.js

```diff
@@ -0,0 +1,121 @@
+import React from 'react';
+import Login from '../login.component';
+import { render, screen, waitFor, userEvent } from '../../../testUtils'
+
+import history from "../../../root/root.history"
+
+import { HOME_PAGE, RESTORE_PASSWORD_PAGE } from "../../../root/root.constants.js";
+
+import { MOCK_CORRECT_USERNAME, MOCK_CORRECT_PASSWORD, MOCK_INCORRECT_USERNAME,
+    MOCK_INCORRECT_PASSWORD, LOGIN_LABEL, PASSWORD_LABEL, LOGIN_BUTTON, FORGOT_PASSWORD
+} from "../__fixtures__/data"
+
+// --- mock AuthService ---
+const mockLoginSuccess = jest.fn()
+const mockLoginFailed = jest.fn()
+
+const mockAuthenticate = (login, password) => {
+    if (login === MOCK_CORRECT_USERNAME && password === MOCK_CORRECT_PASSWORD) {
+        mockLoginSuccess()
+        return Promise.resolve()
+    } else {
+        mockLoginFailed()
+        return Promise.reject({message: "invalid username/password"})
+    }
+}
+
+jest.mock('../../../services/auth.service', () => ({
+    getInstance: jest.fn().mockImplementationOnce(() => ({
+        // FIXME: find a way to mock the userRole dynamically between tests in the same file
```

Comment:
could you have an array like:
userRoles = [ 'field', 'agency', 'global' ]

And put the tests in a for loop?

Or if the different tests expect different results (oh! that's what it's testing! Sorry about my misunderstanding!), an array of documents like:
userRoles = [
{ role: 'agency', expects: [x, y, z] }

??

That could be for another ticket though.

Reply:
Thanks for the suggestion, I gotta admit I haven't tried a for-loop like this.

The challenge is that `jest.mock` is strict on the 1 mock per file thing, and since the `AuthService` class uses static classes, it's even trickier. 
I tried all sorts of workarounds that I can find on Jest documentations, but maybe iterating on `describe` or `test` per role might work.

I'll create a follow-up ticket, thanks!